torch/multiprocessing/spawn.py

```diff
@@ -131,9 +135,9 @@ def join(self, timeout=None):
                 process.terminate()
             process.join()
 
-        # There won't be an error on the queue if the process crashed.
+        # There won't be any content in the error file if the process crashed.
         failed_process = self.processes[error_index]
-        if self.error_queues[error_index].empty():
+        if os.stat(self.error_files[error_index]).st_size == 0:
```

Comment:
In some environments, `/tmp` is cleaned up periodically. If a child process ran longer than the `/tmp` cleanup interval, and it crashed without writing to the error file, it seems that the `os.stat()` here would error out?

Reply:
> The lifetime of these files should be very short (measured in seconds)

IIUC this is true only if the child wrote to the error file, which isn't guaranteed right? The `os.stat()` call here seems to assume the error file to always exist because parent process created it, which can be days/weeks prior to the failure and is suspectible to the cleaners.